src/pybind/mgr/orchestrator.py

```diff
@@ -906,6 +916,17 @@ def __init__(self, name, devices):
     def to_json(self):
         return {'name': self.name, 'devices': [d.to_json() for d in self.devices]}
 
+    @classmethod
+    def from_json(cls, data):
+        try:
+            name = data['name']
+            devices = [InventoryDevice.from_json(device)
+                       for device in data.get('devices', [])]
+            return cls(name, devices)
+        except KeyError as e:
```

Comment:
```suggestion
        except (KeyError, TypeError) as e:
```

Reply:
Fixed. Use a decorator to catch type error (https://github.com/ceph/ceph/pull/29595/commits/cee75dc6e27d213a1048f62e0f429a5317d983a4#diff-dfbc76d00f82320b0e6ac732319aa545R665)

Also update unit tests (https://github.com/ceph/ceph/pull/29595/commits/cee75dc6e27d213a1048f62e0f429a5317d983a4#diff-2abf91c18ca9763c937f021625e81599)